### PR TITLE
Add percent-change histogram to metrics diff report

### PR DESCRIPTION
**Pull request by arnaud-lacurie**
Appends a text histogram of the % change distribution to each field's stats block in the metrics diff report.

- `MetricsStatistics`: collect percent diffs per field in `Builder`, expose as sorted `FieldStatistics.sortedPercentDiffs`
- `MetricsDiffAnalyzer`: render a 2%-wide bin bar chart after each field summary; guard against `lo == hi` edge case (all values identical on a bin boundary)

### CODE REVIEW

File: yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java [hunk 1]
```diff
@@ -541,10 +541,50 @@ private void appendStatisticalSummary(@Nonnull final StringBuilder report, @Nonn
...
+        private void appendHistogram(@Nonnull final StringBuilder report,
+                                     @Nonnull final String fieldName,
+                                     @Nonnull final List<Double> sortedPercentDiffs) {
+            if (sortedPercentDiffs.size() < 3) {
+                return;
+            }
+            final double binWidth = 2.0;
+            final int barWidth = 30;
+            final double lo = Math.floor(sortedPercentDiffs.get(0) / binWidth) * binWidth;
+            double hi = Math.ceil(sortedPercentDiffs.get(sortedPercentDiffs.size() - 1) / binWidth) * binWidth;
```
Comment (robert-brunel) [inline yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java]:
```suggestion
            final double hi = (Math.floor(sortedPercentDiffs.get(sortedPercentDiffs.size() - 1) / binWidth) + 1) * binWidth;
```
That way, the `if (hi <= lo)` check below is not necessary.

File: yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsStatistics.java [hunk 2]
```diff
@@ -165,7 +175,11 @@ private Map<String, FieldStatistics> buildStats(@Nonnull final Map<String, List<
                         .average().orElse(0.0);
                 final var standardDeviation = Math.sqrt(variance);
 
-                builder.put(fieldName, new FieldStatistics(values, mean, standardDeviation));
+                final List<Double> sortedPcts = percentDifferences.getOrDefault(fieldName, ImmutableList.of());
```
Comment (robert-brunel) [inline yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsStatistics.java]: Rather name it just `pcts` instead of `sortedPcts`. At this point it is **not** actually sorted yet!

File: yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java [hunk 3]
```diff
@@ -541,10 +541,50 @@ private void appendStatisticalSummary(@Nonnull final StringBuilder report, @Nonn
...
         }
 
+        private void appendHistogram(@Nonnull final StringBuilder report,
+                                     @Nonnull final String fieldName,
+                                     @Nonnull final List<Double> sortedPercentDiffs) {
+            if (sortedPercentDiffs.size() < 3) {
+                return;
+            }
+            final double binWidth = 2.0;
+            final int barWidth = 30;
```
Comment (robert-brunel) [inline yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java]: Consider turning these into actual arguments (or maybe `private static` constants)?

There’s a comment further down that mentions `BIN_WIDTH`. That would make more sense if there were an actual `private static BIN_WIDTH`.

File: yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java [hunk 4]
```diff
@@ -541,10 +541,50 @@ private void appendStatisticalSummary(@Nonnull final StringBuilder report, @Nonn
...
+            for (final double p : sortedPercentDiffs) {
+                final double b = Math.floor(p / binWidth) * binWidth;
+                bins.merge(b, 1, Integer::sum);
+            }
+
+            final int maxCount = bins.values().stream().mapToInt(Integer::intValue).max().orElse(1);
+
+            report.append(String.format(Locale.ROOT, "`%s` %% change distribution (%d queries, bin = 2%%):%n%n", fieldName, sortedPercentDiffs.size()));
+            report.append("```\n");
+            report.append(String.format(Locale.ROOT, "%14s  %-30s  n%n", "Range", ""));
```
Comment (robert-brunel) [inline yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java]: Is there an (easy) way to avoid hard-coding the `%14s` width here?

File: yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java [hunk 5]
```diff
@@ -541,10 +541,50 @@ private void appendStatisticalSummary(@Nonnull final StringBuilder report, @Nonn
                         report.append("  - No regressions! 🎉\n");
                     }
                     report.append("\n"); // End with blank line
+                    appendHistogram(report, fieldName, fieldStats.sortedPercentDiffs);
                 }
             }
         }
 
+        private void appendHistogram(@Nonnull final StringBuilder report,
```
Comment (robert-brunel) [inline yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MetricsDiffAnalyzer.java:552]: This looks like it can be a `static` method.

It’s a matter of taste, but I would consider splitting off two more submethods:
1. The part that computes the `bins` from `sortedPercentDiffs`.
2. The part that formats a `String` from the `bins`.

These could then be understood and (hypothetically) unit-tested independently.

**Review by RobertBrunel — CHANGES_REQUESTED** [review]

**Review by robert-brunel — APPROVED** [review]
